backend/dal/Helpers/Extensions/ParcelExtensions.cs

```diff
@@ -190,6 +190,19 @@ public static IQueryable<Entity.Parcel> GenerateQuery(this PimsContext context,
             return parcel?.Id;
         }
 
+        /// <summary>
+        /// Get unique a pin that will uniquely identify a parcel using a pre-existing pid.
+        /// </summary>
+        /// <param name="existingPid"></param>
+        /// <returns></returns>
+        public static int GetUniquePidPin(this PimsContext context, int existingPid)
```

Comment:
This should ensure that a new unique PIN is generated when a subdivision is added with a pre-existing PID. Will leave unused pins if a subdivision is deleted, and start counting from a high number if a parcel already has a pin but I think that both of these things should be fine.